src/Illuminate/Database/Eloquent/Relations/HasOneOrMany.php

```diff
@@ -297,6 +297,19 @@ public function create(array $attributes = [])
         });
     }
 
+    /**
+     * Create a new instance of the related model. Allow mass-assignment.
+     *
+     * @param  array  $attributes
+     * @return \Illuminate\Database\Eloquent\Model
+     */
+    public function forceCreate(array $attributes = [])
+    {
+        $attributes[$this->getForeignKeyName()] = $this->getParentKey();
+
+        return $this->related->forceCreate($attributes);
```

Comment:
Hey @atoff , reason I didn't go to that route is because I couldn't find a way to test the unguarded call.

As it is a static method, it is tricky to test it with mockery. Than I decided to keep it simple and preferred to have tests in place, so I deferred to the underlying implementation while adding the parent key.

But if maintainers think your suggestion is better, I am glad to update it.

Reply:
No worries! It also puzzled me a little bit, and it's true that it does make it a bit simpler.

I see this has been merged now, but thought I'd share how I managed to test it incase you are interested! Took me a little bit to work out the mocking too, but I was able to do it by just changing this:
```
protected function expectForceCreatedModel($relation, $attributes)
    {
        $model = $this->expectNewModel($relation, $attributes);
        $relation->getRelated()->shouldReceive('unguarded')->with(Mockery::type(Closure::class))->andReturnUsing(function ($closure) {
            return $closure();
        });

        return $model;
    }
```

Reply:
It worked past the `unguarded` call, I can't recall why it was failing complaining about the method being static to me, maybe it is because of the time I worked on it.

But after using this code, it failed this assertion on the test:

~~~php
$this->assertEquals(1, $created->getAttribute('foreign_key'));
~~~

Which I think is the most important behavior to be tested, as it was what was unexpected before it.

Anyway, thanks a lot for looking into it. The most important is to have it behaving as expected.

If you think the code will be better maintained by refactoring it, you can try a PR. 

As it is still unreleased, I guess there is a chance the maintainers will look over the maintainability benefits, if you describe the advantage, which I can see very clear, well.